lib/aws/rails/notifications_instrument_plugin.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-core'
+
+module Aws
+  module Rails
+    # @api private
+    class NotificationsInstrument < Seahorse::Client::Plugin
+
+      def add_handlers(handlers, config)
+        handlers.add(Handler, step: :initialize)
+      end
+
+      class Handler < Seahorse::Client::Handler
+
+        def call(context)
+          # TODO: Should this be #{operation_name}.#{service}?
+          ActiveSupport::Notifications.instrument('operation.aws_sdk', context: context) do
```

Comment:
"aws.#{client.class}.#{operation_name}"

because notification subscriptions can use regex

Reply:
Yeah good point here.  I went with <operation>.<service>.aws (following the rails convention from https://edgeguides.rubyonrails.org/active_support_instrumentation.html).